spec/models/miq_queue_spec.rb

```diff
@@ -336,6 +336,41 @@
       expect(MiqQueue.get).to eq(nil)
     end
 
+    it "should accept non-Array args (for now)" do
+      begin
+        class MiqQueueSpecNonArrayArgs
+          def self.some_method(single_arg)
+            single_arg
+          end
+        end
+
+        msg = MiqQueue.put(
+          :class_name  => "MiqQueueSpecNonArrayArgs",
+          :method_name => "some_method",
+          :args        => "not_an_array",
+        )
+
+        msg_from_db = MiqQueue.find(msg.id)
+        expect(msg_from_db.args).to eq(["not_an_array"])
+
+        _, _, result = msg_from_db.deliver
+        expect(result).to eq "not_an_array"
+      ensure
+        Object.send(:remove_const, :MiqQueueSpecNonArrayArgs)
```

Comment:
I refuse to add a silly before/after block in this case because I'd need to add a whole context / before / after just for this... to a file with a billion before/context blocks.  

Sorry RSpec, your its are :hankey: for this... mintest beats you on this:

```ruby
def test_something
  ...
ensure
  Object.send(:remove_const, :MiqQueueSpecNonArrayArgs)
end
```